youtube_dl/extractor/npo.py

```diff
@@ -1,767 +1,387 @@
+class HetKlokhuisIE(NTRSubsiteIE):
+    IE_NAME = 'hetklokhuis'
+    IE_DESC = 'hetklokhuis.nl'
+    _VALID_URL = r'https?://(?:www\.)?hetklokhuis\.nl/.*'
+    _TESTS = [{
+        'url': 'https://hetklokhuis.nl/dossier/142/zoek-het-uit/tv-uitzending/2987/aliens',
+        'md5': '4664b54ed4e05183b1e4f2f4290d551e',
         'info_dict': {
-            'id': 'WO_NTR_429477',
-            'display_id': 'ademhaling-de-hele-dag-haal-je-adem-maar-wat-gebeurt-er-dan-eigenlijk-in-je-lichaam',
-            'title': 'Ademhaling: De hele dag haal je adem. Maar wat gebeurt er dan eigenlijk in je lichaam?',
+            'id': 'aliens',
+            'title': 'aliens',
             'ext': 'mp4',
-            'description': 'md5:abfa0ff690adb73fd0297fd033aaa631'
         },
-        'params': {
-            # Skip because of m3u8 download
-            'skip_download': True
-        }
-    }
+    }]
 
 
-class HetKlokhuisIE(NPODataMidEmbedIE):
-    IE_NAME = 'hetklokhuis'
-    _VALID_URL = r'https?://(?:www\.)?hetklokhuis\.nl/[^/]+/\d+/(?P<id>[^/?#&]+)'
-
-    _TEST = {
-        'url': 'http://hetklokhuis.nl/tv-uitzending/3471/Zwaartekrachtsgolven',
+class VPROIE(NPOIE):
+    IE_NAME = 'vpro'
+    IE_DESC = 'vpro.nl'
+    _VALID_URL = r'https?://(?:www\.)?vpro.nl/.*'
+    _TESTS = [{
+        'url': 'https://www.vpro.nl/programmas/tegenlicht/kijk/afleveringen/2015-2016/offline-als-luxe.html',
+        'md5': 'cf302e066b5313cfaf8d5adf50d64f13',
         'info_dict': {
-            'id': 'VPWON_1260528',
-            'display_id': 'Zwaartekrachtsgolven',
-            'ext': 'm4v',
-            'title': 'Het Klokhuis: Zwaartekrachtsgolven',
-            'description': 'md5:c94f31fb930d76c2efa4a4a71651dd48',
-            'upload_date': '20170223',
+            'id': 'offline-als-luxe.html',
+            'title': 'offline-als-luxe.html',
+            'ext': 'mp4',
         },
-        'params': {
-            'skip_download': True
-        }
-    }
-
+    }]
 
-class NPOPlaylistBaseIE(NPOIE):
     def _real_extract(self, url):
-        playlist_id = self._match_id(url)
-
-        webpage = self._download_webpage(url, playlist_id)
-
-        entries = [
-            self.url_result('npo:%s' % video_id if not video_id.startswith('http') else video_id)
-            for video_id in orderedSet(re.findall(self._PLAYLIST_ENTRY_RE, webpage))
-        ]
-
-        playlist_title = self._html_search_regex(
-            self._PLAYLIST_TITLE_RE, webpage, 'playlist title',
-            default=None) or self._og_search_title(webpage)
-
-        return self.playlist_result(entries, playlist_id, playlist_title)
+        video_id = url.rstrip('/').split('/')[-1]
+        page = self._download_webpage(url, video_id)
+        results = re.findall(r'data-media-id="([a-zA-Z0-9_]+)"\s', page)
+        formats = []
+        for result in results:
+            formats.extend(self._extract_formats_by_product_id(result, video_id))
+            break
```

Comment:
If the desired behaviour is to use just the first match, this is easier: 

```suggestion
        page = self._download_webpage(url, video_id)
        format = traverse_obj(re.search(r'data-media-id="([a-zA-Z0-9_]+)"\s', page), (
                1, T(lambda x: self._extract_formats_by_product_id(x, video_id)),
                Ellipsis))
```
Or if not getting `page` shouldn't crash:
```suggestion
        page = self._download_webpage(url, video_id, fatal=False) or ''
        format = traverse_obj(re.search(r'data-media-id="([a-zA-Z0-9_]+)"\s', page), (
            1, T(lambda x: self._extract_formats_by_product_id(x, video_id)),
            Ellipsis))
```
To be extra sure (if the target must be an attribute value of an HTML element), you can search for elements like `r'<[\w.-]+\s[^>]*(?<!-)\bdata-media-id\s*=\s*[^>]+>' and then get the first or all `extract_attributes(element).get('data-media-id') that is not `None`.